set.mm

```diff
@@ -647669,29 +647787,355 @@ standardize vectors to a length (norm) of one, but such definitions require
+      VLUUKUQAUUMUURYKGVOZAUUNUURYKIVOZBDUAVPVQVRUUTUVKUVFUAQUUTUVAQSZUMZUVDUVJ
+      UVEUVPUVDUVJUVPUVDUMZUVHUVIUVQUVHUVAMNOZXTTRZUVQUVRYEXTTUVQUVRYBYDUVQUVRU
+      VQUVAUUTUVOUVDVSZUNVDZUVQUVCUVRYBTRZUVPUVBUVCVTUVQUVOUUQUVCUWBUQUVTUUTUUQ
+      UVOUVDAUUPUUQYKWEZVOZUVAYAWAVQWBZUVQUVBUVRYDTRZUVPUVBUVCWCUVQUVOUUPUVBUWF
+      UQUVTUUTUUPUVOUVDAUUPUUQYKWFZVOUVAYCWAVQWBZUVQYBUVQYAUWDUNVDZUVQYDUUTYDQS
+      UVOUVDUUTYCUWGUNZVOVDZWGUUTYFUVOUVDUUSYFYGYJWHZVOWIUVQUVOUUMUVHUVSUQUVTUU
+      TUUMUVOUVDUVMVOUVABWAVQVRUVQUVIUVRYHTRZUVQUVRYIYHTUVQUVRYBYDUWAUWEUWHUWIU
+      WKWJUUTYJUVOUVDUUSYFYGYJWKZVOWIUVQUVOUUNUVIUWMUQUVTUUTUUNUVOUVDUVNVOUVADW
+      AVQVRWLWNWMWOVFUUTUUPUUQUVGYMUQUWGUWCYCYAUAVPVQWBUUTYNYBYHUIRUUTYBYHYDUCO
+      YHUIUUTYHYBYDUUTYAUUTYAUWCVGZWPZUUTYCUUTYCUWGVGZWPZUWNWQUUTYHYDUUTYHAYTUU
+      RYKUUFVOWRUUTYDUWJXDWSWTUUTYADUUTYAUWCWRUUTDUVNWRUUTYAUUTYAUWCXAXBUUTDUUT
+      DUVNXAXBXCVRUUTYQYOYPUUTYQUUTYAUKUWCUKXESUUTXFUSZXGVGUUTYOUUTYCUKUWGUWSXG
+      VGUUTYIYEUDOZYEYIUDOZYQYOUCOZYPUUTYIYEUUTYBYDUWPUWRXHUUTYBYDUWPUWRXIXJUUT
+      UXBYBMNOZYDMNOZUCOZUWTUUTYQUXCYOUXDUCUUTYAUWOVHUUTYCUWQVHXKUUTYBXOSYDXOSU
+      XEUWTPUWPUWRYBYDXLVQXMUUTYPXTYHUDOUXAUUTBDUUTBUVMVGUUTDUVNVGXPUUTXTYEYHYI
+      UDUWLUWNXKXMXNXQXRWNXSVF $.
+  $}
+
+  ${
+    $d A i $.  $d B i $.  $d C i $.  $d ph i $.
+    flt4lem6b.a $e |- ( ph -> A e. NN ) $.
+    flt4lem6b.b $e |- ( ph -> B e. NN ) $.
+    flt4lem6b.c $e |- ( ph -> C e. NN ) $.
+    flt4lem6b.1 $e |- ( ph -> 2 || A ) $.
+    flt4lem6b.2 $e |- ( ph -> ( A gcd C ) = 1 ) $.
+    flt4lem6b.3 $e |- ( ph -> ( ( A ^ 2 ) + ( B ^ 2 ) ) = ( C ^ 2 ) ) $.
+    $( Construct a smaller counterexample when ` A ` is even.  (Contributed by
+       SN, 21-Aug-2024.) $)
+    flt4lem6b $p |- ( ph -> ? ) $=
+      ? $.
+  $}
+
+  ${
+    flt4lem6.a $e |- ( ph -> A e. NN ) $.
+    flt4lem6.b $e |- ( ph -> B e. NN ) $.
+    flt4lem6.c $e |- ( ph -> C e. NN ) $.
+    flt4lem6.2 $e |- ( ph -> ( A gcd C ) = 1 ) $.
+    flt4lem6.3 $e |- ( ph -> ( ( A ^ 4 ) + ( B ^ 2 ) ) = ( C ^ 4 ) ) $.
+    $( Given a solution to ` A ^ 4 + B ^ 2 = C ^ 4 ` , provide a smaller
+       solution.  This satisfies the infinite descent condition.  (Contributed
+       by SN, 20-Aug-2024.) $)
+    flt4lem6 $p |- ( ph -> E. z e. NN ( E. d e. NN E. e e. NN (
+         ( d gcd z ) = 1 /\ ( ( d ^ 4 ) + ( e ^ 2 ) ) = ( z ^ 4 ) ) /\
+         z < C ) $=
+      ? $.
+  $}
+
+  ${
+    $d ph a c $.  $d ph b $.  $d A a $.  $d A c $.  $d B a $.  $d B b c $.
+    $d C c $.  $d a b d e f $.  $d c d e f $.  $d e g h $.   $d e f x z $.
+    $d ph d g h x z $.
+    flt4g.a $e |- ( ph -> A e. NN ) $.
+    flt4g.b $e |- ( ph -> B e. NN ) $.
+    flt4g.c $e |- ( ph -> C e. NN ) $.
+    $( Fermat's last theorem for the exponent four, generalized.  This is
+       equivalent to Fermat's right triangle theorem.  (Contributed by SN,
+       20-Aug-2024.) $)
+    flt4g $p |- ( ph -> ( ( A ^ 4 ) + ( B ^ 2 ) ) =/= ( C ^ 4 ) ) $=
```

Comment:
Is it legitimate to call this a "generalization of Fermat's last theorem for the exponent four"?

Reply:
Yes, by taking the contrapositive of $A^4 + B^4 = C^4 \implies A^4 + (B^2)^2 = C^4$

(Note: New proof, this will become $A^4 + B^4 = C^2$ instead)

Reply:
It is indeed stronger, but "generalization" is a bit far-fetched. Maybe:
```
$( Fermat's right triangle theorem, which implies Fermat's last theorem for exponent 4.
   (Contributed by SN, 20-Aug-2024.) $)
```
And the "new proof" should actually be another theorem: both theorems should be here.  The other comment can be
```
$( Strengthening of Fermat's last theorem for exponent 4, where the sum is
   only assumed to be a square.  (Contributed by SN, 20-Aug-2024.) $)
```

Reply:
I see you did that. Thanks. Resolving.